### DISCUSSION TITLE
A node that can communicate with the cluster but is not a member of the cluster

### DISCUSSION THREAD

**Discussion by civilizeddev** [Tochemey/goakt#207] [Q&A] [answered]
I want to know if this is possible in **Goakt**.

To explain, it is as follows: Members of a goakt cluster are hosting actors on it. The question is whether there is a way for an external client, who is not a member of this cluster but can communicate with this actor system, to send messages to the actors.

![image](https://www.plantuml.com/plantuml/svg/XSx12i8m383XUvuYxAIm7Tmk8cFi1Rpv0hKBOvYseqb14D_TJdSOUDYb3JyV9BseYPeS0qIoQlp8X8-pu0iGBOSqv3U9KloMt8ARWoUlM3XoTue56i7VUDeLzQveidW3R97bQVdQLgRbF0s-wo2RTIcsLRMQDTN_gTaalE2Dc0zMdoNEqr3YbKJCI4U0dloG77y0)

For reference, in the case of **protoactor**, the clusterprovier offers `StartMember()` to start as a member of the cluster and `StartClient()` to start merely as a client, separately.

https://github.com/asynkron/protoactor-go/blob/2ecba7517929a27dc1ac0c44734b9c538b9044cb/cluster/clusterproviders/k8s/k8s_provider.go#L101-L129

I would like to know if this is also possible in Goakt. If there is no direct method, I wonder if there might be an indirect way.

**Comment by civilizeddev** [comment]
This seems to be related to the PR. I need a usage example regarding **eventstream**.

https://github.com/Tochemey/goakt/pull/18

**Reply by Tochemey** [reply]
@civilizeddev I have some examples in the actor-cluster folder that show how to communicate with actors in the cluster. Using the [api](https://github.com/Tochemey/goakt/blob/main/actors/api.go) one can communicate to an actor from an external system. I hope this answers your questions

**Reply by Tochemey** [reply]
You can see the examples here:

- https://github.com/Tochemey/goakt/blob/main/examples/actor-cluster/k8s/service/service.go#L76
- https://github.com/Tochemey/goakt/blob/main/examples/actor-cluster/k8s/service/service.go

**Reply by Tochemey** [reply]
Regarding the event stream, only deadletter are emitted because events are occurrences happened in the actor system that can be propagated via a pubsub mechanism. So far only deadletter are propagated. In a nutshell it is internal to Go-Akt at the moment.

**Reply by Tochemey** [reply]
I will be enhancing the internal events in this issue #171

**Comment by civilizeddev** [comment]
> You can see the examples here:
> 
> * https://github.com/Tochemey/goakt/blob/main/examples/actor-cluster/k8s/service/service.go#L76
> * https://github.com/Tochemey/goakt/blob/main/examples/actor-cluster/k8s/service/service.go

In this example, the service is running on the same node as the actor system. Does this mean that in this example, I can create an ActorSystem without starting and use it in the service? Is that really possible?

**Reply by Tochemey** [reply, marked as answer]
@civilizeddev From what you are asking the only apis you can use are:

- https://github.com/Tochemey/goakt/blob/main/actors/api.go#L191
- https://github.com/Tochemey/goakt/blob/main/actors/api.go#L228
- https://github.com/Tochemey/goakt/blob/main/actors/api.go#L268
- https://github.com/Tochemey/goakt/blob/main/actors/api.go#L306
- https://github.com/Tochemey/goakt/blob/main/actors/api.go#L350

Those one do not require an actor system running on your system. However you need an actor system running on the remote node you are making the requests to. I hope this answers your question.

**Reply by Tochemey** [reply]
> > You can see the examples here:
> 
> > 
> 
> > * https://github.com/Tochemey/goakt/blob/main/examples/actor-cluster/k8s/service/service.go#L76
> 
> > * https://github.com/Tochemey/goakt/blob/main/examples/actor-cluster/k8s/service/service.go
> 
> 
> 
> In this example, the service is running on the same node as the actor system. Does this mean that in this example, I can create an ActorSystem without starting and use it in the service? Is that really possible?

No that is possible 😉

**Reply by civilizeddev** [reply]
So, you're saying that the ActorSystem is referenced only as a source of information for remote calls? I understand.

Then, can the client immediately decide which pod to make a remote call to?

**Reply by Tochemey** [reply]
The actor system in those examples help create and locate actors locally and remotely.
@civilizeddev please what do you mean by the client?
